### PR TITLE
Add `trusted` query param. to `incoming-transfers` endpoint

### LINKED ISSUE #1206: Add `trusted` query to `incoming-transfers` endpoint
### Description

The `incoming-transfers` endpoint does not include the `trusted` query that the `history` endpoint does.

Web uses the above endpoint when filtering transactions. Without the query, [untrusted transactions are shown when filtering](https://github.com/safe-global/safe-wallet-web/issues/3334).

### Requirements

- Add the `trusted` query to the `incoming-transfers` route
- Abstract and implement the [`onlyTrusted` filtering](https://github.com/safe-global/safe-client-gateway/blob/3832c65975edd0b258e0b4bee2548ed1c9267f7b/src/routes/transactions/mappers/transactions-history.mapper.ts#L194-L209) to the route

### Additional information

There was [a request](https://github.com/safe-global/safe-wallet-web/issues/3334#issuecomment-1964083328) to implement the above on the `multisig-transactions` and `module-transactions` endpoints but `trusted` only affects `incoming-transfers`.

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This resolves #1206 by adding the `trusted` query param. from the `history` endpoint to the `incoming-transfers` one. The associated logic has been moved from`TransactionsHistoryMapper['mapTransfers']` into a new `TransferMapper['mapTransfers']` method that accepts an `onlyTrusted` argument.

## Changes

- A `trusted` query param to `TransactionsController['getIncomingTransfers']`
- Move trust-related filtering from `TransactionsHistoryMapper['mapTransfers']` to `TransferMapper['mapTransfers']`
- Migrate all usage of `TransferMapper['mapTransfer']` to `TransferMapper['mapTransfers']`
- Add/updated associated tests.

### CODE REVIEW

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: src/routes/transactions/transactions.service.ts [hunk 1]
```diff
@@ -268,17 +269,14 @@ export class TransactionsService {
...
-          ),
-      ),
+    const results = await this.transferMapper.mapTransfers({
+      chainId: args.chainId,
+      transfers: transfers.results,
+      safe: safeInfo,
+      onlyTrusted: args.onlyTrusted,
+    });
+    const incomingTransfers = results.map(
+      (incomingTransfer) => new IncomingTransfer(incomingTransfer),
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/routes/transactions/transactions.service.ts]: Changed in 3e8a4b2a7748e3ea848cc83e59ecd71381d4248b.

File: src/routes/transactions/mappers/transfers/transfer.mapper.spec.ts [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+      const safe = safeBuilder().build();
+      const addressInfo = new AddressInfo(faker.finance.ethereumAddress());
+      addressInfoHelper.getOrDefault.mockResolvedValue(addressInfo);
+
+      const actual = await mapper.mapTransfer(chainId, transfer, safe);
+
+      expect(actual).toBeInstanceOf(Transaction);
+      expect(actual).toEqual(
+        expect.objectContaining({
+          id: `${TRANSFER_PREFIX}${TRANSACTION_ID_SEPARATOR}${safe.address}${TRANSACTION_ID_SEPARATOR}${transfer.transferId}`,
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/routes/transactions/mappers/transfers/transfer.mapper.spec.ts]: I adjusted it in 3e8a4b2a7748e3ea848cc83e59ecd71381d4248b to not use `objectContaining` but still `expect.any(TrasferTransactionInfo` for the `txInfo` as that is covered by the `transfer-info.mapper.spec.ts`.

File: src/routes/transactions/__tests__/controllers/list-incoming-transfers-by-safe.transactions.controller.spec.ts [hunk 3]
```diff
@@ -263,6 +264,146 @@ describe('List incoming transfers by Safe - Transactions Controller (Unit)', ()
...
+                },
+              },
+              conflictType: 'None',
+            },
+          ],
+        });
+      });
+  });
+
+  it('Should filter out non-trusted ERC20 incoming transfers', async () => {
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/routes/transactions/__tests__/controllers/list-incoming-transfers-by-safe.transactions.controller.spec.ts]: I clarified the description in 3e8a4b2a7748e3ea848cc83e59ecd71381d4248b.

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**